src/operations/blas2/trsv.hpp

```diff
@@ -0,0 +1,142 @@
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *  SYCL-BLAS: BLAS implementation using SYCL
+ *
+ *  @filename trsv.hpp
+ *
+ **************************************************************************/
+
+#ifndef TBSV_HPP
+#define TBSV_HPP
+#include "operations/blas2_trees.h"
+#include "operations/blas_operators.hpp"
+#include "views/view_sycl.hpp"
+#include <stdexcept>
+#include <vector>
+namespace blas {
+
+/**
+ * @struct Trsv
+ * @brief Tree node representing a triangular band matrix_ vector_
+ * multiplication.
+ */
+template <typename lhs_t, typename matrix_t, typename vector_t,
+          uint32_t local_range, bool is_upper, bool is_transposed,
+          bool is_unitdiag>
+SYCL_BLAS_INLINE
+Trsv<lhs_t, matrix_t, vector_t, local_range, is_upper, is_transposed,
+     is_unitdiag>::Trsv(lhs_t &_l, matrix_t &_matrix,
+                        typename Trsv<lhs_t, matrix_t, vector_t, local_range,
+                                      is_upper, is_transposed,
+                                      is_unitdiag>::index_t &_blk_id,
+                        vector_t &_vector)
+    : lhs_(_l), matrix_(_matrix), vector_(_vector), blk_id_(_blk_id) {}
+
+template <typename lhs_t, typename matrix_t, typename vector_t,
+          uint32_t local_range, bool is_upper, bool is_transposed,
+          bool is_unitdiag>
+SYCL_BLAS_INLINE typename Trsv<lhs_t, matrix_t, vector_t, local_range, is_upper,
+                               is_transposed, is_unitdiag>::index_t
+Trsv<lhs_t, matrix_t, vector_t, local_range, is_upper, is_transposed,
+     is_unitdiag>::get_size() const {
+  return matrix_.get_size();
+}
+template <typename lhs_t, typename matrix_t, typename vector_t,
+          uint32_t local_range, bool is_upper, bool is_transposed,
+          bool is_unitdiag>
+SYCL_BLAS_INLINE bool
+Trsv<lhs_t, matrix_t, vector_t, local_range, is_upper, is_transposed,
+     is_unitdiag>::valid_thread(cl::sycl::nd_item<1> ndItem) const {
+  // Valid threads are established by ::eval.
+  return true;
+}
+
+template <typename lhs_t, typename matrix_t, typename vector_t,
+          uint32_t local_range, bool is_upper, bool is_transposed,
+          bool is_unitdiag>
+template <typename local_memory_t>
+SYCL_BLAS_INLINE typename Trsv<lhs_t, matrix_t, vector_t, local_range, is_upper,
+                               is_transposed, is_unitdiag>::value_t
+Trsv<lhs_t, matrix_t, vector_t, local_range, is_upper, is_transposed,
+     is_unitdiag>::eval(local_memory_t local_mem, cl::sycl::nd_item<1> ndItem) {
+  const index_t _offset = blk_id_ * local_range;
+
+  const index_t g_idx = _offset + ndItem.get_local_id(0);
+  const index_t l_idx = ndItem.get_local_id(0);
+
+  const index_t _N = lhs_.get_size();
+  auto l_x = local_mem.localAcc;
```

Comment:
A suggestion here, maybe we could change `l_x` to `scratch_acc` or something.

Reply:
I would like to keep a descriptive name: `l_x` signals the fact this is a portion of the local solution.